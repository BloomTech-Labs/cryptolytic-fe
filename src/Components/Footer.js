import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Link, Grid } from "@material-ui/core";
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles(t => ({
	footer: {
		backgroundColor: "black",
		display: "flex",
		justifyContent: "flex-end",
		alignItems: "center",
		height: "10vh",
		width: "90%",
		padding: '60px 0px 225px 0px',

	},
	footerNavigation: {
		zIndex: 1,
		backgroundColor: "black",
		color: "#f0f8ff",
		display: "flex",
		alignItems: "center",
		width: "70%",
		height: "100%",
		paddingTop: 30,
		marginTop: 150
	},
	footerLinks: {
		fontFamily: 'Titillium Web, sans-serif',
		fontStyle: 'normal',
		fontWeight: 600,
		fontSize: '1.6rem',
		lineHeight: '49px',
		display: 'flex',
		alignItems: 'center',
		textAlign: 'center',
		margin: '5px',
		'&:hover': {
			textDecoration: 'none',
			color: 'gray'
		}
	},
	footerTitle: {
		fontFamily: 'Titillium Web, sans-serif',
		fontStyle: 'normal',
		fontWeight: 'bold',
		fontSize: '3rem',
		lineHeight: '76px',
		display: 'flex',
		alignItems: 'center',
		textAlign: 'center',
		letterSpacing: '0.03em'
	},
	footerBody: {
		fontFamily: 'Titillium Web, sans-serif',
		fontStyle: 'normal',
		fontWeight: 600,
		fontSize: '1.6rem',
		lineHeight: '49px',
		display: 'flex',
		alignItems: 'center',
		textAlign: 'center'
	},
	footerCopyright: {
		fontFamily: 'Titillium Web, sans-serif',
		fontStyle: 'normal',
		fontWeight: 'normal',
		fontSize: '1.4rem',
		lineHeight: '37px',
		display: 'flex',
		alignItems: 'center',
		textAlign: 'center',
		marginLeft: '1vw'
	}
}));

const Footer = () => {
	const classes = useStyles();
	return (
		<div className={classes.footer}>
			<nav className={classes.footerNavigation}>
				<Grid container spacing={0}>
					<Grid
						item
						sm={4}
						direction='column'
						container
						alignItems='flex-start'
					>
						<Typography className={classes.footerTitle}>Cryptolytic</Typography>
						<div style={{ margin: '30px 0px' }}>
							{['Contact Us', 'cryptolyticapp@gmail.com'].map(e =>
								<Typography className={classes.footerBody}>{e}</Typography>
							)}
						</div>
						<Typography className={classes.footerCopyright}>Copyright 2020 Cryptolytic</Typography>
					</Grid>
					<Grid item sm={4} container direction='column' alignItems='center'>
						{['Home', 'About', 'Services'].map(e =>
							<Link color='inherit' className={classes.footerLinks}>{e}</Link>
						)}
					</Grid>
					<Grid item sm={3} container direction='column' alignItems='center'>
						{['Legal', 'Privacy Policy', 'User Agreement', 'FAQ'].map(e =>
							<Link color='inherit' className={classes.footerLinks}>{e}</Link>
						)}
					</Grid>
				</Grid>
			</nav>
		</div>
	);
};

export default Footer;
