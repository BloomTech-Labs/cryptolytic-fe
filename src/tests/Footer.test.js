import React from "react";
///
import Footer from "../Components/Footer";
///
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

test("Footer component initializes without crashing", () => {
  render(<Footer />);
});

describe("<Footer/> matches verified SnapShot", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
