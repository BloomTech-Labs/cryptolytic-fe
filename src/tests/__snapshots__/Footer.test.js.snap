// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> matches verified SnapShot should match snapshot 1`] = `
<div
  className="makeStyles-footer-146"
>
  <nav
    className="makeStyles-footerNavigation-147"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-grid-sm-4"
      >
        <p
          className="MuiTypography-root makeStyles-footerTitle-149 MuiTypography-body1"
        >
          Cryptolytic
        </p>
        <div
          style={
            Object {
              "margin": "30px 0px",
            }
          }
        >
          <p
            className="MuiTypography-root makeStyles-footerBody-150 MuiTypography-body1"
          >
            Contact Us
          </p>
          <p
            className="MuiTypography-root makeStyles-footerBody-150 MuiTypography-body1"
          >
            cryptolyticapp@gmail.com
          </p>
        </div>
        <p
          className="MuiTypography-root makeStyles-footerCopyright-151 MuiTypography-body1"
        >
          Copyright 2020 Cryptolytic
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-grid-sm-4"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-footerLinks-148 MuiTypography-colorInherit"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Home
        </a>
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-footerLinks-148 MuiTypography-colorInherit"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          About
        </a>
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-footerLinks-148 MuiTypography-colorInherit"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Services
        </a>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-grid-sm-3"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-footerLinks-148 MuiTypography-colorInherit"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Legal
        </a>
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-footerLinks-148 MuiTypography-colorInherit"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Privacy Policy
        </a>
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-footerLinks-148 MuiTypography-colorInherit"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          User Agreement
        </a>
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-footerLinks-148 MuiTypography-colorInherit"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          FAQ
        </a>
      </div>
    </div>
  </nav>
</div>
`;
